File: aoc-2015/src/test/java/com/capital7software/aoc2015/days/Day05Test.java
package com.capital7software.aoc2015.days;

import static org.junit.jupiter.api.Assertions.assertEquals;

import com.capital7software.aoc.aoc2015.days.Day05;
import com.capital7software.aoc.lib.AdventOfCodeTestBase;
import java.util.List;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

class Day05Test extends AdventOfCodeTestBase {
  private static final Logger LOGGER = LoggerFactory.getLogger(Day05Test.class);

  @BeforeEach
  void setUp() {
    var instance = new Day05();
    setupFromFile(instance.getDefaultInputFilename());
  }

  @Test
  void getNaughtyOrNice() {
    var instance = new Day05();
    var expected = List.of(true, true, false, false, false);
    var expectedCount = 2;
    var actualCount = 0;
    for (int i = 0; i < lines.size(); i++) {
      var actual = instance.isNice(lines.get(i));

      if (actual) {
        actualCount++;
      }
      assertEquals(expected.get(i), actual, "Test input: " + lines.get(i));
    }
    assertEquals(expectedCount, actualCount);
  }

  @Test
  void getNewNaughtyOrNice() {
    setupFromFile("inputs/input_day_05-02.txt");
    var instance = new Day05();
    var expected = List.of(true, true, false, false);
    var expectedCount = 2;
    var actualCount = 0;
    for (int i = 0; i < lines.size(); i++) {
      var actual = instance.isNewNice(lines.get(i));

      if (actual) {
        actualCount++;
      }
      assertEquals(expected.get(i), actual, "Test input: " + lines.get(i));
    }
    assertEquals(expectedCount, actualCount);
  }

  @Override
  protected Logger getLogger() {
    return LOGGER;
  }
}

File: aoc-2024/src/test/kotlin/com/capital7software/aoc/aoc2024aoc/days/Day05Test.kt
package com.capital7software.aoc.aoc2024aoc.days

import com.capital7software.aoc.lib.AdventOfCodeTestBase
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class Day05Test : AdventOfCodeTestBase() {
  companion object {
    val log: Logger = LoggerFactory.getLogger(Day05Test::class.java)
  }

  override fun getLogger(): Logger {
    return log
  }

  @BeforeEach
  fun setup() {
    val instance = Day05()
    setupFromFile(instance.defaultInputFilename)
  }

  @Test
  fun testCalculateOccurrencesOfXmas() {
    val instance = Day05()
    val expected = 143L

    val actual = instance.sumOfValidMiddlePages(lines)
    Assertions.assertEquals(
        expected,
        actual,
        "The actual sum of $actual is not the expected sum of $expected."
    )
  }

  @Test
  fun testSumOfInvalidMiddlePages() {
    val instance = Day05()
    val expected = 123L

    val actual = instance.sumOfInvalidMiddlePages(lines)
    Assertions.assertEquals(
        expected,
        actual,
        "The actual sum of $actual is not the expected sum of $expected."
    )
  }
}
